File: ros2_ws/src/example_python/example_python/sub_node.py
import rclpy
from rclpy.node import Node 
from interfaces.msg import Example

class SubNode(Node):

    def __init__(self):
        super().__init__('pub_node')
        self.subsciber = self.create_subscription(Example, "ExampleTopic", self.subscriber_callback, 10)

    def subscriber_callback(self, message: Example):
        print(message)

def main():
    rclpy.init()
    node = SubNode()
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        pass
    finally:
        node.destroy_node()
        rclpy.shutdown()

if __name__ == '__main__':
    main()

File: ros2_ws/src/example_cpp/src/sub_node.cpp
#include "interfaces/msg/example.hpp"
#include "rclcpp/rclcpp.hpp"

class SubNode : public rclcpp::Node
{
public:
    SubNode() : Node("sub_node")
    {
        subscriber_ = this->create_subscription<interfaces::msg::Example>("ExampleTopic", 10, std::bind(&SubNode::subscriber_callback, this, std::placeholders::_1));
    }

private:
    void
    subscriber_callback(const interfaces::msg::Example::SharedPtr example)
    {
        RCLCPP_INFO(this->get_logger(), "Recieving: %i", example->message);
    }
    rclcpp::Subscription<interfaces::msg::Example>::SharedPtr subscriber_;
};

int main(int argc, char *argv[])
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<SubNode>());
    rclcpp::shutdown();
    return 0;
}